packages/sdk/src/lib/dataProtectorCore/processProtectedData.ts

```diff
@@ -263,6 +279,45 @@ export const processProtectedData = async ({
       isDone: true,
     });
 
+    // Handle result encryption
+    let privateKey: string | undefined;
+    if (vEncryptResult) {
+      const { publicKey, privateKey: generatedPrivateKey } =
+        await getFormattedKeyPair({
+          pemPrivateKey: vPemPrivateKey,
+        });
+      privateKey = generatedPrivateKey;
+
+      // Notify user if a new key was generated
+      if (!vPemPrivateKey) {
+        vOnStatusUpdate({
+          title: 'PUSH_ENCRYPTION_KEY',
+          isDone: false,
+          payload: {
+            message:
+              'New encryption key pair generated and stored in IndexedDB',
+          },
+        });
```

Comment:
fix the message and pass the generated key

```suggestion
        vOnStatusUpdate({
          title: 'GENERATE_ENCRYPTION_KEY',
          isDone: true,
          payload: {
            message:
              'New encryption key pair generated',
            pemPrivateKey: generatedPrivateKey,
          },
        });
```